Source/Download.swift

```diff
@@ -150,8 +150,8 @@ extension Request {
         // MARK: Delegate Methods
 
         func URLSession(session: NSURLSession, downloadTask: NSURLSessionDownloadTask, didFinishDownloadingToURL location: NSURL) {
-            if downloadTaskDidFinishDownloadingToURL != nil {
-                let destination = downloadTaskDidFinishDownloadingToURL!(session, downloadTask, location)
+            if let handler = downloadTaskDidFinishDownloadingToURL {
+                let destination = handler(session, downloadTask, location)
```

Comment:
In all the places where you are replacing force unwrapping with optional binding, could you use the following format instead?

```swift
if let downloadTaskDidFinishDownloadingToURL = self. downloadTaskDidFinishDownloadingToURL {
    let destination = downloadTaskDidFinishDownloadingToURL(session, downloadTask, location)
}
```

We have a open task for replacing all property accesses with the `self.` prefix for clarity. This would be a good place to start that work on all these changes.